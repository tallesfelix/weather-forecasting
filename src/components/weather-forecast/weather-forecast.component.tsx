import { LocationAutocomplete } from '../location-autocomplete/location-autocomplete.component';
import { WeatherCard } from '../weather-card/weather-card.component';
import styles from './weather-forecast.module.css';

type Coordinates = {
  latitude: number;
  longitude: number;
}

type WeatherToday = {
  conditions: string;
  tempmin: number;
  tempmax: number;
  date: string;
  icon: string;
}

type WeatherForecastProps = {
  setLocation: React.Dispatch<React.SetStateAction<Coordinates>>;
  weather: WeatherToday[];
}

export const WeatherForecast = ({ weather, setLocation }: WeatherForecastProps) => {
  return (
    <section className={styles.footer}>
      <div className={styles.footerContainer}>
        <LocationAutocomplete setLocation={setLocation} />
        <ul className={styles.timesForecasting}>
          {weather && weather.map((day) => (
            <li key={day.date}>
              <WeatherCard
                conditions={day.conditions}
                tempMin={day.tempmin}
                tempMax={day.tempmax}
                date={day.date}
                icon={day.icon}
              />
            </li>
          ))}
        </ul>
      </div>
    </section>
  );
};