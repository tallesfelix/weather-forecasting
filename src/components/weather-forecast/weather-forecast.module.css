.footer {
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  height: 300px;
  box-shadow: 0px -13px 22px 3px rgba(0, 0, 0, 0.1);
  padding: 16px;
  flex-shrink: 0;
}

.footerContainer {
  max-width: 780px;
  margin: 0 auto;
  text-align: left;
  position: relative;
}

.timesForecasting {
  display: flex;
  list-style: none;
  justify-content: space-between;
  gap: 10px;
  margin-top: 8px;

  & li {
    min-height: 178px;
  }
}