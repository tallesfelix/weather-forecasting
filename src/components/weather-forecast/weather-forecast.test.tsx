import { render, screen } from '@testing-library/react';
import { WeatherForecast } from './weather-forecast.component';

describe('WeatherForecast', () => {
  test('renders WeatherForecast component', () => {
    const { asFragment } = render(<WeatherForecast
      setLocation={jest.fn()}
      weather={[
        {
          conditions: 'Snow',
          tempmin: 0,
          tempmax: 4,
          date: '2021-01-01',
          icon: 'snow'
        }
      ]}
    />);
    expect(screen.getByText(/Snow/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
