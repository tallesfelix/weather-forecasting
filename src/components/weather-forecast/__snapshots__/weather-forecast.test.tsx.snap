// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherForecast renders WeatherForecast component 1`] = `
<DocumentFragment>
  <section
    class="footer"
  >
    <div
      class="footerContainer"
    >
      <div
        class="container"
      >
        <input
          class="searchInput"
          disabled=""
          placeholder="Search for another location"
          type="text"
          value=""
        />
      </div>
      <ul
        class="timesForecasting"
      >
        <li>
          <div
            class="weatherCard"
          >
            <div>
              <img
                alt="snow"
                src=""
              />
              <p>
                Snow
              </p>
            </div>
            <div>
              <span>
                min. 0°C
              </span>
              <span>
                max. 4°C
              </span>
              <p>
                1/1
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
